src/main/java/baritone/utils/schematic/format/defaults/LitematicaSchematic.java

```diff
@@ -0,0 +1,296 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Baritone.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package baritone.utils.schematic.format.defaults;
+
+import baritone.utils.schematic.StaticSchematic;
+import baritone.utils.accessor.INBTTagLongArray;
+import net.minecraft.block.*;
+import net.minecraft.block.properties.IProperty;
+import net.minecraft.nbt.*;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.block.state.IBlockState;
+
+import org.apache.commons.lang3.Validate;
+import javax.annotation.Nullable;
+import java.util.*;
+
+/**
+ * @author Emerson
+ * @since 12/27/2020
+ * @author rycbar
+ * @since 22.09.2022
+ *
+ */
+public final class LitematicaSchematic extends StaticSchematic {
+    int minX=0;
+    int minY=0;
+    int minZ=0;
+    private static final String reg = "Regions";
+    private static final String meta = "Metadata";
+    private static final String schemSize = "EnclosingSize";
+    private static final String blSt = "BlockStates";
+    private static final String blStPl = "BlockStatePalette";
+    private static final String pos = "Position";
+    private static final String size = "Size";
+
+    public LitematicaSchematic(NBTTagCompound nbt) {
+        getMinimumCorner(nbt);
+
+        this.x = Math.abs(nbt.getCompoundTag(meta).getCompoundTag(schemSize).getInteger("x"));
+        this.y = Math.abs(nbt.getCompoundTag(meta).getCompoundTag(schemSize).getInteger("y"));
+        this.z = Math.abs(nbt.getCompoundTag(meta).getCompoundTag(schemSize).getInteger("z"));
+        this.states = new IBlockState[this.x][this.z][this.y];
+
+        for (String subReg : getRegions(nbt)) {
+            NBTTagList usedBlockTypes = nbt.getCompoundTag(reg).getCompoundTag(subReg).getTagList(blStPl, 10);
+            IBlockState[] blockList = getBlockList(usedBlockTypes);
+
+            int bitsPerBlock = getBitsPerBlock(usedBlockTypes.tagCount());
+            long regionVolume = getVolume(nbt, subReg);
+            long[] blockStateArray = getBlockStates(nbt, subReg);
+
+            LitematicaBitArray bitArray = new LitematicaBitArray(bitsPerBlock, regionVolume, blockStateArray);
+
+            writeSubregionIntoSchematic(nbt, subReg, blockList, bitArray);
+        }
+    }
+
+    /**
+     * @return Array of subregion names.
+     */
+    private static String[] getRegions(NBTTagCompound nbt) {
+        return nbt.getCompoundTag(reg).getKeySet().toArray(new String[0]);
+    }
+
+    /**
+     * Calculates the minimum cords/origin of the schematic as litematica schematics
+     * can have a non-minimum origin.
+     */
+    private void getMinimumCorner(NBTTagCompound nbt) {
+        for (String subReg : getRegions(nbt)) {
+            this.minX = Math.min(this.minX, getMinimumCord(nbt, subReg,"x"));
```

Comment:
this feels gross. modifying minx like this is weird. can we instead calculate these mins "once and for all" in the constructor then save them in a final field?